pipeline/src/test_util.rs

```diff
@@ -62,27 +77,37 @@ pub fn regular_test_small_field(file_name: &str, inputs: &[i32]) {
 /// Tests witness generation, mock prover, pilcom and plonky3 with BabyBear.
 pub fn regular_test_bb(file_name: &str, inputs: &[i32]) {
     let inputs_bb = inputs.iter().map(|x| BabyBearField::from(*x)).collect();
-    let pipeline_bb = make_prepared_pipeline(file_name, inputs_bb, vec![]);
+    // LinkerMode::Native because the bus is not implemented for small fields
+    let pipeline_bb = make_prepared_pipeline(file_name, inputs_bb, vec![], LinkerMode::Native);
     test_mock_backend(pipeline_bb.clone());
     test_plonky3_pipeline(pipeline_bb);
 }
 
 /// Tests witness generation, mock prover, pilcom and plonky3 with BabyBear and KoalaBear.
 pub fn regular_test_kb(file_name: &str, inputs: &[i32]) {
     let inputs_kb = inputs.iter().map(|x| KoalaBearField::from(*x)).collect();
-    let pipeline_kb = make_prepared_pipeline(file_name, inputs_kb, vec![]);
+    // LinkerMode::Native because the bus is not implemented for small fields
+    let pipeline_kb = make_prepared_pipeline(file_name, inputs_kb, vec![], LinkerMode::Native);
     test_mock_backend(pipeline_kb.clone());
     test_plonky3_pipeline(pipeline_kb);
 }
 
 /// Tests witness generation, mock prover, pilcom and plonky3 with Goldilocks.
 pub fn regular_test_gl(file_name: &str, inputs: &[i32]) {
-    let inputs_gl = inputs.iter().map(|x| GoldilocksField::from(*x)).collect();
-    let pipeline_gl = make_prepared_pipeline(file_name, inputs_gl, vec![]);
-
-    test_mock_backend(pipeline_gl.clone());
-    run_pilcom_with_backend_variant(pipeline_gl.clone(), BackendVariant::Composite).unwrap();
-    test_plonky3_pipeline(pipeline_gl);
+    let inputs_gl = inputs
+        .iter()
+        .map(|x| GoldilocksField::from(*x))
+        .collect::<Vec<_>>();
+
+    let pipeline_gl_native =
+        make_prepared_pipeline(file_name, inputs_gl.clone(), vec![], LinkerMode::Native);
+    test_mock_backend(pipeline_gl_native.clone());
+    run_pilcom_with_backend_variant(pipeline_gl_native, BackendVariant::Composite).unwrap();
+
+    let pipeline_gl_bus =
```

Comment:
Note that this means we now run steps such as constant generation, witgen, etc twice, once for each linker mode.